tests/installation/snapper_rollback_test.pm

```diff
@@ -0,0 +1,72 @@
+# SUSE's openQA tests
+#
+# Copyright © 2009-2013 Bernhard M. Wiedemann
+# Copyright © 2012-2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+
+# the test's goal is to test a rollback (as a backup) after system has migrated.
+# we make one rollback to going back to downgraded system, (1)
+# and we make another rollback to go to upgraded system. at the end 6 snapshots.
+# we are in a read_only_snapshot 1 and after we are in a read_and_write 2
+
+
+# TODO: make cleanup test, to remove the 4 created snapshots? inside or outside
+#       testsuite?. to discuss.
+use base "consoletest";
+use testapi;
+use utils;
+
+
+sub run() {
+    my $self = shift;
+    #1) read only snapshot. Don't try to save something here.
+    script_run("snapper rollback -d rollback-before-migration");
+
+    assert_script_run("snapper list | tail -n 2 | grep rollback");
+
+    # we are in the before migration version. So snapshot 4 is identical to
+    # snapshot 1.
+    # 1 before migration = 4 since we did not already booted on snapshot 4.
+    # After normal_booting in 4 there is login stuff that differs from 1.
+    script_run("echo \"SNAPSHOT_1&4_IDENTICAL\" >/dev/$serialdev", 0);
+    script_run("snapper diff 1..4 >/dev/$serialdev",               0);
+    script_run("snapper status 1..4>/dev/$serialdev",              0);
+    # FIXME if test passed, $serialdev should be empty ( maybe to new lines \n?)
+    wait_serial('^SNAPSHOT_1&4_IDENTICAL') || die "snapshots differ when they should have been identical";
```

Comment:
Ah, so you are testing if `snapper diff` and `snapper status` add something to serial output. Shouldn't you then check for something like `wait_serial('^SNAPSHOT_1&4_IDENTICAL[\n]{3}$')`?